src/meta/src/barrier/mod.rs

```diff
@@ -651,10 +663,19 @@ where
                 new_epoch,
                 prev_epoch
             );
-            state
+
+            // rollback when failed to update meta-storage.
+            if let Err(e) = state
```

Comment:
I think this will be timeout in most cases, and we should always retry here? We may only trigger recovery for those unrecoverable errors. 🤔

Reply:
How about adding the retry logic to the meta store implementation? cc @TennyZhuang @yezizp2012

Reply:
+1 for adding retry logic to meta store implementation, this might happen in most cases when read or write on meta store, not only in `update_inflight_prev_epoch`.